### ISSUE TITLE
GraphQL Asset transforms incorrectly return upscaled image dimensions

### ISSUE THREAD

**Issue by FreekVR** [craftcms/cms#6084]
### Description
Given the following asset transformation fragment;
```gql
fragment menuImage on AssetInterface {
    title
    height @transform(mode: "fit", height: 400, width: 380)
    width @transform(mode: "fit", height: 400, width: 380)
    url @transform(mode: "fit", height: 400, width: 380, immediately: true)
}
```
the config setting `upscaleImages => false`,
and an image source file that was originally 240x240px

I would expect the following:
* The image is not really resized, retaining its dimensions, because "fit" should not upscale images
* The returned width and height is thus 240x240

What is actually happening:
* The image correctly retains its original dimensions when transformed
* The returned width/height are `380x380`, as if the image has actually been upscaled.

Eg. what I'm left with is a 240x240 image which will be rendered as 380x380 in my frontend :(

### Additional info

- Craft version: 3.4.16
- PHP version: 7.2
- Database driver & version: MySQL 5.7
- Plugins & versions: Amazon S3 1.2.7

**Comment by andris-sevcenko** [comment]
Just fixed this for the next Craft 3.4.x release!

**Comment by brandonkelly** [comment]
Craft 3.4.21 is out now with that fix.

**Comment by FreekVR** [comment]
I'm afraid something is still awry here, but I'm not sure if it's related.

Given an input image of 240x240px

and a transform as follows (upscaleImages being off):
```
    height @transform(mode: "crop", width: 340, height: 300, position: "top-center")
    width @transform(mode: "crop", width: 340, height: 300, position: "top-center")
    url @transform(mode: "crop", width: 340, height: 300, immediately: true, position: "top-center")
```

The image is cropped to `240x122` matching the transform definition aspect ratio.

However the returned values for both `width` and `height` are now `240`, as in the source image, resulting in an incorrect aspect ratio when using these properties in the `<img>`-element.

**Comment by andris-sevcenko** [comment]
Just fixed this (again) for the next Craft 3.4 release.

**Comment by brandonkelly** [comment]
Craft 3.4.25 is out now with the fix.
